content/en/docs/how-to/registry-authentication.md

```diff
@@ -17,7 +17,69 @@ To authenticate to an OCI-compliant registry, use the `notation login` command w
 notation login -u <username> -p <password> <registry>
 ```
 
-> **Note:** If `notation login` is failing, you may need to configure [Docker Credential Store](https://github.com/notaryproject/notation/blob/main/specs/registry-auth.md#credential-store) for your environment.
+> **Note:** If `notation login` is failing, you may need to configure [Docker Credential Store](https://github.com/notaryproject/notation/blob/main/specs/registry-auth.md#credential-store) for your environment or configure environment variables as detailed in the *Configure Docker Credential Store for Linux* section.
+
+## Configure environment variables to authenticate to an OCI-compliant registry
+
+Set the `NOTATION_USERNAME` and `NOTATION_PASSWORD` environment variables to authenticate to an OCI-compliant registry.
+
+```console
+export NOTATION_USERNAME="YOUR_REGISTRY_USERNAME"
+export NOTATION_PASSWORD="YOUR_REGISTRY_PASSWORD"
+```
+
+After the environment variables are set, you can use `notation login` without specifying a username or password.
+
+For security reasons, unset the environment variables after you log out from the registry. For example:
+
+```console
+unset NOTATION_USERNAME
+unset NOTATION_PASSWORD
+```
+
+## Configure Docker Credential Store for Linux
+
+As a security best practice, you should use a credential helper with a system keychain when using local credentials to access remote repositories.
+
+`notation` uses [Docker Credential Helpers](https://github.com/docker/docker-credential-helpers) and its [protocol](https://docs.docker.com/engine/reference/commandline/login/#credential-helper-protocol) as the credential helpers. Currently, using Docker Credential Helpers requires manual installation and configuration in Linux by following the steps below. This configuration will be simplified in Notation v1.0.0.
+
+Install [Docker credential helper](https://github.com/docker/docker-credential-helpers) pass.
+
+```console
+mkdir -p ~/bin
+curl -Lo ~/bin/docker-credential-pass "https://github.com/docker/docker-credential-helpers/releases/download/v0.7.0/docker-credential-pass-v0.7.0.linux-amd64"
+chmod +x ~/bin/docker-credential-pass
+```
+
+Generate and configure GPG key for encryption. 
+
+> **NOTE:** If you have a GPP key already, run `gpg --edit-key` to trust your key instead.
+
+```console
+gpg --full-generate-key
+```
+
+Install and configure [pass](https://www.passwordstore.org/).
+
+```console
+sudo apt update 
+sudo apt install pass -y
+pass init $your_email
+```
+
+Configure Docker credential store in `/.docker/config.json`.
+
+```
+mkdir -p ~/.docker
+echo '{"credsStore":"pass"}' > ~/.docker/config.json
+chmod 600 ~/.docker/config.json
+```
+
+Use `notation login` to authenticate to an OCI-compliant registry.
+
+```console
+notation login -u <username> -p <password> <registry>
```

Comment:
Passwords should be typed via stdin, and should not be passed as a command line argument.
```suggestion
notation login -u <username> <registry>
```
`-p` is there just for dev / testing purposes.

The security behind it is that any executed shell command might be logged somewhere in plaintext.

Reply:
Agree. I realized that the login prompt feature is supported in RC.4.